plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BaseBigQueryTypeMapping.java

```diff
@@ -526,44 +533,128 @@ public void testTime()
+    }
+
+    private SqlDataTypeTest testTimestampWithTimeZone(String inputType)
+    {
+        return SqlDataTypeTest.create()
                 // min value in BigQuery
-                .addRoundTrip("TIMESTAMP", "TIMESTAMP '0001-01-01 00:00:00.000000 UTC'",
+                .addRoundTrip(inputType, "TIMESTAMP '0001-01-01 00:00:00.000 UTC'",
                         TIMESTAMP_TZ_MICROS, "TIMESTAMP '0001-01-01 00:00:00.000000 UTC'")
-                .addRoundTrip("TIMESTAMP", "TIMESTAMP '1970-01-01 00:00:00.000000 UTC'",
+                // before epoch
+                .addRoundTrip(inputType, "TIMESTAMP '1958-01-01 13:18:03.123 UTC'",
+                        TIMESTAMP_TZ_MICROS, "TIMESTAMP '1958-01-01 13:18:03.123000 UTC'")
+                // after epoch
+                .addRoundTrip(inputType, "TIMESTAMP '2019-03-18 10:01:17.987 UTC'",
+                        TIMESTAMP_TZ_MICROS, "TIMESTAMP '2019-03-18 10:01:17.987000 UTC'")
+                .addRoundTrip(inputType, "TIMESTAMP '2018-10-28 01:33:17.456 UTC'",
+                        TIMESTAMP_TZ_MICROS, "TIMESTAMP '2018-10-28 01:33:17.456000 UTC'")
+                .addRoundTrip(inputType, "TIMESTAMP '2018-10-28 03:33:33.333 UTC'",
+                        TIMESTAMP_TZ_MICROS, "TIMESTAMP '2018-10-28 03:33:33.333000 UTC'")
+                // epoch
+                .addRoundTrip(inputType, "TIMESTAMP '1970-01-01 00:00:00.000 UTC'",
                         TIMESTAMP_TZ_MICROS, "TIMESTAMP '1970-01-01 00:00:00.000000 UTC'")
-                .addRoundTrip("TIMESTAMP", "TIMESTAMP '1970-01-01 00:00:00.000000 Asia/Kathmandu'",
-                        TIMESTAMP_TZ_MICROS, "TIMESTAMP '1969-12-31 18:30:00.000000 UTC'")
-                .addRoundTrip("TIMESTAMP", "TIMESTAMP '1970-01-01 00:00:00.000000+02:17'",
-                        TIMESTAMP_TZ_MICROS, "TIMESTAMP '1969-12-31 21:43:00.000000 UTC'")
-                .addRoundTrip("TIMESTAMP", "TIMESTAMP '1970-01-01 00:00:00.000000-07:31'",
-                        TIMESTAMP_TZ_MICROS, "TIMESTAMP '1970-01-01 07:31:00.000000 UTC'")
-                .addRoundTrip("TIMESTAMP", "TIMESTAMP '1958-01-01 13:18:03.123456 UTC'",
+                .addRoundTrip(inputType, "TIMESTAMP '1970-01-01 00:13:42.000 UTC'",
+                        TIMESTAMP_TZ_MICROS, "TIMESTAMP '1970-01-01 00:13:42.000000 UTC'")
+                .addRoundTrip(inputType, "TIMESTAMP '2018-04-01 02:13:55.123 UTC'",
+                        TIMESTAMP_TZ_MICROS, "TIMESTAMP '2018-04-01 02:13:55.123000 UTC'")
+                .addRoundTrip(inputType, "TIMESTAMP '2018-03-25 03:17:17.000 UTC'",
+                        TIMESTAMP_TZ_MICROS, "TIMESTAMP '2018-03-25 03:17:17.000000 UTC'")
+                .addRoundTrip(inputType, "TIMESTAMP '1986-01-01 00:13:07.000 UTC'",
+                        TIMESTAMP_TZ_MICROS, "TIMESTAMP '1986-01-01 00:13:07.000000 UTC'")
+
+                // same as above but with higher precision
```

Comment:
`2021-09-07 23:59:59.999999` doesn't exist in the above block. You can just remove this comment. I don't think it's helpful for readers.